Weater-app/src/components/customButton/CustomButton.jsx

```diff
@@ -1,49 +1,61 @@
 import React from 'react';
 import { useDispatch, useSelector } from 'react-redux';
+import { ThemeProvider } from 'styled-components';
 
 import {
-  changeLang,
-  changeModal,
-  changeTheme,
+  togglehModal,
+  toggleLang,
+  toggleTheme,
 } from '../../store/thunk/action/app';
+import { languegeObject } from '../../store/data/languageObject';
 
 import dropIcon from '../../assets/image/dropIcon.png';
 import russiaFlagIcon from '../../assets/image/russiaFlagIcon.png';
 import unitedKingdomFlagIcon from '../../assets/image/unitedKingdomFlagIcon.png';
 
-import { BtnWrapper } from './styledComponents/CustomButton';
+import {
+  blackTheme,
+  lightTheme,
+  BtnWrapper,
+} from './styledComponents/CustomButton';
 
 const CustomButton = () => {
-  const theme = useSelector((state) => state.app.theme);
+  const state = useSelector((state) => state.app);
+  const { isBlackTheme, lang } = state;
   const dispatch = useDispatch();
 
   return (
-    <BtnWrapper changeTheme={theme}>
-      <button className="btn" onClick={() => dispatch(changeModal())}>
-        history
-      </button>
-      <div className="imageCircle">
-        <img
-          src={dropIcon}
-          alt="dropIcon"
-          onClick={() => dispatch(changeTheme())}
-        />
-      </div>
-      <div className="imageCircle">
-        <img
-          src={russiaFlagIcon}
-          alt="russiaFlagIcon"
-          onClick={() => dispatch(changeLang('ru'))}
-        />
-      </div>
-      <div className="imageCircle">
-        <img
-          src={unitedKingdomFlagIcon}
-          alt="unitedKingdomFlagIcon"
-          onClick={() => dispatch(changeLang('eng'))}
-        />
-      </div>
-    </BtnWrapper>
+    <ThemeProvider theme={isBlackTheme ? blackTheme : lightTheme}>
+      <BtnWrapper>
+        <button className="btn" onClick={() => dispatch(togglehModal())}>
+          {languegeObject[lang].historyBtn}
```

Comment:
can we swith to solution i suggested? so this line will look like 
```
 <button className="btn" onClick={() => dispatch(togglehModal())}>
{t('History')}
</button>
```

it will make code easier to read (no need to open dictionary and find translation by code + we don't need to use `lang` everywhere